src/middleware.ts

```diff
@@ -5,9 +5,15 @@ import { getToken } from "next-auth/jwt";
 export async function middleware(req: NextRequest) {
   const token = await getToken({ req });
   const isAuth = !!token;
+  const isSignedUp =
+    (token?.profile as { firstName: string | null })?.firstName !== null &&
+    (token?.profile as { lastName: string | null })?.lastName !== null;
+
```

Comment:
Once you change the types then you shouldn't have to do this anymore.

For firstname and lastname, the type can also be undefined, you can check if its falsy using `!!token?.profile?.firstName`

Also add a member condition `token.profile.member !== null`

Reply:
Also maybe rename isSignedUp -> isOnboarded?